chatterbot/adapters/adaptation.py

```diff
@@ -0,0 +1,35 @@
+from chatterbot.utils.module_loading import import_module
+
+
+class Adaptation(object):
+    """
+    This is a base from which ChatterBot inherits utility methods
+    and a context attribute that allows access of adapters to be
+    shared between other adapters. This also makes it possible to
+    share other context information such as a name, or the current
+    conversation with each of the adapters.
+    """
+
+    class context(object):
```

Comment:
Because both of these are static, any changes made to the class of the `data` property within it will be made across all instances. If you are looking for the context to be instance-specific, you need to create a new instance of the class and assign it as a attribute during initialization.

```python
class B:
    def __init__(self):
        self.x = 0

class A:
    def __init__(self):
        self.b = B()
```

```python
>>> a = A()
>>> b = A()
>>> print(a.b.x)
0
>>> print(b.b.x)
0
>>> a.b.x = 1
>>> b.b.x = 2
>>> print(a.b.x)
1
>>> print(b.b.x)
2
```

Note that if you made the `x` attribute static as well, you would face a similar issue within the nested class.